### PR TITLE
feature: add the handling of duplicate project names 

### PR DESCRIPTION
**Pull request by Nirvana-Jie**
## Summary

image

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [x] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run changeset`.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/create-rspack/index.js [hunk 1]
```diff
@@ -4,34 +4,51 @@ const { hideBin } = require("yargs/helpers");
...
+			root = path.resolve(process.cwd(), targetDir);
+		}
+		fs.mkdirSync(root, { recursive: true });
+		// TODO support more template in the future
+		const templateDir = "template-react";
+		const srcFolder = path.resolve(__dirname, templateDir);
+		copyFolder(srcFolder, targetDir);
+		console.log("\nDone. Now run:\n");
+		console.log(`cd ${targetDir}\n`);
+		console.log(`npm install\n`);
```
Comment (h-a-n-a) [inline packages/create-rspack/index.js:50]: Don't miss this: https://github.com/web-infra-dev/rspack/pull/2146

Comment (Nirvana-Jie) [reply packages/create-rspack/index.js:50]: OK,I will change it soon

Comment (h-a-n-a) [reply packages/create-rspack/index.js:50]: Would you reset the whole commit and force push it ? The other contributor failed to sign-up the CLA

Comment (Nirvana-Jie) [reply packages/create-rspack/index.js:50]: OK, I will

**Pull request closed without merging**